src/Compilers/Core/Portable/Operations/OperationExtensions.cs

```diff
@@ -322,5 +322,51 @@ internal static IOperation GetRootOperation(this IOperation operation)
 
             return operation;
         }
+
+        /// <summary>
+        /// Gets either a loop or a switch operation that corresponds to the given branch operation.
+        /// </summary>
+        /// <param name="operation">The branch operation for which a corresponding operation is looked up</param>
+        /// <returns>The corresponding operation or <c>null</c> in case not found (e.g. no loop or switch syntax)</returns>
+        /// <exception cref="ArgumentNullException"><paramref name="operation"/> is null</exception>
+        /// <exception cref="InvalidOperationException">The operation is a part of Control Flow Graph or it has an invalid branch kind. 
+        /// Applicable kinds: <see cref="BranchKind.Break"/> and <see cref="BranchKind.Continue"/>.</exception>
+        public static IOperation GetCorrespondingOperation(this IBranchOperation operation)
+        {
+            if (operation == null)
+            {
+                throw new ArgumentNullException(nameof(operation));
+            }
+
+            if (operation.SemanticModel == null)
+            {
+                throw new InvalidOperationException(CodeAnalysisResources.OperationMustNotBeControlFlowGraphPart);
+            }
+            
+            if (operation.BranchKind != BranchKind.Break && operation.BranchKind != BranchKind.Continue)
+            {
+                throw new InvalidOperationException(
```

Comment:
I think we want to return `null` here instead of throwing. We can potentially extend this method in future to return the target [ICaseClauseOperation](http://source.roslyn.io/#Microsoft.CodeAnalysis/Operations/ICaseClauseOperation.cs,17) for [BranchKind.GoTo](http://source.roslyn.io/#Microsoft.CodeAnalysis/Operations/BranchKind.cs,28)

Reply:
Agreed.

---
In reply to: [208969708](https://github.com/dotnet/roslyn/pull/28285#discussion_r208969708) [](ancestors = 208969708)